ush/setup.py

```diff
@@ -37,384 +41,199 @@
-            or (len(LSM_SPP_TSCALE) != N_VAR_LNDP)
-        ):
+
+    # Check that input dates are in a date format
+    dates = ["DATE_FIRST_CYCL", "DATE_LAST_CYCL"]
+    for val in dates:
+        if not isinstance(cfg_d["workflow"][val], datetime.date):
             raise Exception(
-                f"""
-                All Noah or RUC-LSM SPP-related namelist variables (except ISEED_LSM_SPP)
-                set in {EXPT_CONFIG_FN} must be equal in number of entries to what is found in
-                SPP_VAR_LIST:
-                  LSM_SPP_VAR_LIST (length {len(LSM_SPP_VAR_LIST)})
-                  LSM_SPP_MAG_LIST (length {len(LSM_SPP_MAG_LIST)})
-                  LSM_SPP_LSCALE (length {len(LSM_SPP_LSCALE)})
-                  LSM_SPP_TSCALE (length {len(LSM_SPP_TSCALE)})
-                """
+                dedent(
+                    f"""
+                            Date variable {val}={cfg_d['user'][val]} is not in a valid date format.
+
+                            For examples of valid formats, see the Users' Guide.
+                            """
+                )
             )
-    #
-    # The current script should be located in the ush subdirectory of the
-    # workflow directory.  Thus, the workflow directory is the one above the
-    # directory of the current script.
-    #
-    HOMEdir = os.path.abspath(os.path.dirname(__file__) + os.sep + os.pardir)
 
-    #
-    # -----------------------------------------------------------------------
-    #
-    # Set the base directories in which codes obtained from external reposi-
-    # tories (using the manage_externals tool) are placed.  Obtain the rela-
-    # tive paths to these directories by reading them in from the manage_ex-
-    # ternals configuration file.  (Note that these are relative to the lo-
-    # cation of the configuration file.)  Then form the full paths to these
-    # directories.  Finally, make sure that each of these directories actu-
-    # ally exists.
-    #
-    # -----------------------------------------------------------------------
-    #
-    mng_extrns_cfg_fn = os.path.join(HOMEdir, "Externals.cfg")
+
+    return cfg_d
+
+
+def set_srw_paths(ushdir, expt_config):
+
+    """
+    Generate a dictionary of directories that describe the SRW
+    structure, i.e., where SRW is installed, and the paths to
+    external repositories managed via the manage_externals tool.
+
+    Other paths for SRW are set as defaults in config_defaults.yaml
+
+    Args:
+       ushdir:      (str) path to the system location of the ush/ directory
+                     under the SRW clone
+       expt_config: (dict) contains the configuration settings for the
+                     user-defined experiment
+
+    Returns:
+       dictionary of config settings and system paths as keys/values
+    """
+
+    # HOMEdir is the location of the SRW clone, one directory above ush/
+    home_dir = os.path.abspath(os.path.dirname(__file__) + os.sep + os.pardir)
```

Comment:
Can we use `homedir` instead of `home_dir` just for the sake of rename_model script?

Reply:
Yep!